subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -920,6 +929,12 @@ static int tls_mbedtls_init(struct tls_context *context, bool is_server)
 		return ret;
 	}
 
+	ret = mbedtls_ssl_conf_alpn_protocols(&context->config,
```

Comment:
Needs an `#ifdef CONFIG_MBEDTLS_SSL_ALPN`,
did not notice as I had it enabled when working on it :)